upi/vsphere/masters/main.tf

```diff
@@ -0,0 +1,81 @@
+data "vsphere_datacenter" "dc" {
+  name = "${var.datacenter}"
+}
+
+data "vsphere_compute_cluster" "compute_cluster" {
+  name          = "${var.cluster}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
+
+data "vsphere_datastore" "datastore" {
+  name          = "${var.datastore}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
+
+data "vsphere_network" "network" {
+  name          = "${var.vm_network}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
+
+data "vsphere_virtual_machine" "template" {
+  name          = "${var.vm_template}"
+  datacenter_id = "${data.vsphere_datacenter.dc.id}"
+}
+
+resource "vsphere_resource_pool" "resource_pool" {
+  name                    = "${var.resource_pool}"
+  parent_resource_pool_id = "${data.vsphere_compute_cluster.compute_cluster.resource_pool_id}"
+}
+
+resource "vsphere_virtual_machine" "vm" {
+  count            = 4
+  name             = "ocp${count.index + 1}"
+  resource_pool_id = "${vsphere_resource_pool.resource_pool.id}"
+  datastore_id     = "${data.vsphere_datastore.datastore.id}"
+  num_cpus         = 4
+  memory           = 8192
```

Comment:
making the master cpus and memory configurable with a variable would be nice.

Reply:
Agreed! Parameterized CPU and memory and set these as defaults in variables.tf